File: 프로그래머스/lv1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cpp
using namespace std;

long long solution(int price, int money, int count)
{
    long long answer;
    long long sum;
    
    sum = count * (count + 1) / 2;
    answer = sum  * price;
    
    if (answer >= money) answer -= money;
    else answer = 0;
    
    return answer;
}

File: 프로그래머스/lv1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cs
using System;

class Solution
{
    public long solution(int price, int money, int count)
    {
        long answer = 0;
		for(int i = 0; i<count; i++)
		{
			answer += (i+1) * price;
		}
		if (answer >= money) answer -= money;
		else answer = 0;
		return answer;
    }
}